client/context/tx_generator.go

```diff
@@ -0,0 +1,51 @@
+package context
+
+import (
+	"github.com/tendermint/tendermint/crypto"
+
+	"github.com/cosmos/cosmos-sdk/types"
+)
+
+type (
+	// TxGenerator defines an interface a client can utilize to generate an
+	// application-defined concrete transaction type. The type returned must
+	// implement TxBuilder.
+	TxGenerator interface {
```

Comment:
Being that we're using `any`, what is the need or purpose of `TxGenerator` now?

Reply:
It lets us have backwards compatibility with amino `StdTx` which is actually what's happening in this PR. Which seems like it's valuable at least short term. So in #6216, I'm intending to enable the protobuf tx pathway and have proto/amino integration tests working side by side.

Reply:
But aren't we still using a `StdTx` with `any` or are we using an interface?

Reply:
There's the amino `StdTx` uses `sdk.Msg` interfaces and then there's the new proto `Tx` we added to `types/tx`